pkg/cmd/dockerregistry/dockerregistry.go

```diff
@@ -137,7 +138,15 @@ func Execute(configFile io.Reader) {
 	app.Config.HTTP.Headers.Set("X-Registry-Supports-Signatures", "1")
 
 	app.RegisterHealthChecks()
-	handler := alive("/", app)
+	handler := http.Handler(app)
```

Comment:
I've updated this PR, it still has one TODO, but the main idea I think will stay the same: use two limiters, one for GET/HEAD requests, and another one for PATCH/PUT and mirroring.